### DISCUSSION TITLE
C-h/j/k/l cant be overwritten any longer? or existing tmux.nvim keymaps stopped working

### DISCUSSION THREAD

**Discussion by madhermit** [LazyVim/LazyVim#5126] [Q&A] [answered]
Hey all,

With one of the recent updates (i think the very first snacks related one?) my tmux.nvim plugin keymaps stopped working:

```lua
return {
  "aserowy/tmux.nvim",
  event = "VeryLazy",
  config = function()
    require("tmux").setup()
  end,
  keys = {
    { "<leader>h", "<cmd>lua require('tmux').move_left()<cr>", desc = "Go to left window" },
    { "<C-h>", "<cmd>lua require('tmux').move_left()<cr>", desc = "Go to left window" },
    { "<C-j>", "<cmd>lua require('tmux').move_bottom()<cr>", desc = "Go to lower window" },
    { "<C-k>", "<cmd>lua require('tmux').move_top()<cr>", desc = "Go to upper window" },
    { "<C-l>", "<cmd>lua require('tmux').move_right()<cr>", desc = "Go to right window" },
  },
}
```
the `<leader>h` keymap works, and running `lua require('tmux').move_left()` directly works, so i know its not the plugin itself.

putting the keymaps into `lua/config/keymaps.lua` doesn't work either:

```lua
-- Navigate between tmux panes and vim splits
vim.keymap.set("n", "<C-h>", "<Cmd>lua require'tmux'.move_left()", {})
...
```

Any ideas?

Thanks in advance!

**Comment by madhermit** [comment]
Any thoughts on this? Its driving me crazy.

Thanks!

**Reply by dpetka2001** [reply]
I don't use this plugin, but in my old configuration I used `christoomey/vim-tmux-navigator` and it worked as expected. See [here](https://github.com/dpetka2001/dotfiles/blob/dcfe546f46758dfd8821058b843884f114fc4c9a/dot_config/nvim/lua/config/keymaps.lua#L83-L88) how I had configured my keybindings. And in `tmux.conf` I only had [this](https://github.com/dpetka2001/dotfiles/blob/dcfe546f46758dfd8821058b843884f114fc4c9a/dot_config/tmux/tmux.conf#L75-L76). 

I just tested it now and it still works. I use Kitty as my terminal.

**Reply by madhermit** [reply]
Thanks for the response! I tried switching over to `christoomey/vim-tmux-navigator` but its the same issue.
Because the `<leader>h` remap works with both plugins, I'm assuming it is something outside those plugins that's preventing the remap of `<C-h|j|k|l>` , I event tried mapping to `<C-;>`, so maybe its something with the ctrl key? `<S-h>` does in fact work.

**Reply by dpetka2001** [reply]
Probably something related to your terminal and how the Control modifier is being translated by tmux. Try Kitty terminal and see if it works for you there.

**Reply by dpetka2001** [reply]
In Kitty i also had to enable these [settings in tmux](https://github.com/dpetka2001/dotfiles/blob/dcfe546f46758dfd8821058b843884f114fc4c9a/dot_config/tmux/tmux.conf#L28-L30) for Control modifiers to work.

**Reply by madhermit** [reply]
Thanks for all the ideas. Turns out the problem is indeed with Tmux. If I try without tmux it works just fine. Confirmed this behavior with Ghostty, Wezterm & Kitty. Unfortunately, the tmux settings you mentioned didn't work, even for Kitty.
Odd that `Ctrl-w h` works in neovim and `<C-h>`  worked until just recently.

**Reply by dpetka2001** [reply]
Strange because it works for me that I tested with Kitty. I don't know what to tell you. What version of tmux are you using?

Here's a screencast of how it looks like for me

[Screencast_2025-01-17-19-40-54.webm](https://github.com/user-attachments/assets/2a897c5b-b789-4a0d-a4e9-169128aeb55c)

**Reply by madhermit** [reply]
Since it happened recently it must be related to a version number. I'm running `tmux 3.5`, how about you?
I did find this ticket on `vim-tmux-navigator`  [Switching tmux panes with C-h broken with tmux master branch](https://github.com/christoomey/vim-tmux-navigator/issues/421)

**Reply by madhermit** [reply]
> I don't use this plugin, but in my old configuration I used `christoomey/vim-tmux-navigator` and it worked as expected.


BTW what are you using for your new configuration?

**Reply by dpetka2001** [reply]
I'm using `tmux 3.5a`.

**Reply by dpetka2001** [reply]
I don't use tmux any more. I'm just complacent with Neovim tabs, Snacks terminal and also Kitty tabs. I'm still new to programming as I started at the beginning of this year, so I don't have complicated workflow where I need to do many multiple things. So, the above combination is enough for me.

**Comment by madhermit** [comment, marked as answer]
Upgrading to tmux 3.5a fixed my problem
